zone/gm_commands/parcels.cpp

```diff
@@ -0,0 +1,299 @@
+			database,
+			fmt::format("to_name = '{}' ORDER BY slot_id ASC", player_name.c_str())
+		);
+
+		if (results.empty()) {
+			c->Message(Chat::White, fmt::format("No parcels could be found for {}", player_name).c_str());
+			return;
+		}
+
+		c->Message(Chat::Yellow, fmt::format("Found {} parcels for {}.", results.size(), player_name).c_str());
+
+		for (auto const &p: results) {
+			c->Message(
+				Chat::Yellow,
+				fmt::format(
+					"Slot [{:02}] has item id [{:10}] with quantity [{}].",
+					p.slot_id,
+					p.item_id,
+					p.quantity
+				).c_str()
+			);
+		}
+	}
+	if (is_listmemory) {
+		auto player_name = std::string(sep->arg[2]);
+		auto player      = entity_list.GetClientByName(player_name.c_str());
+
+		if (arguments < 2) {
+			c->Message(Chat::White, "Usage: #parcels listmemory [Character Name]");
+		}
+
+		if (!player) {
+			c->Message(
+				Chat::White,
+				fmt::format(
+					"Player {} could not be found in this zone.  Ensure you are in the same zone as the player.",
+					player_name
+				).c_str()
+			);
+			return;
+		}
+
+		auto parcels = player->GetParcels();
+		if (parcels.empty()) {
+			c->Message(Chat::White, fmt::format("No parcels could be found for {}", player_name).c_str());
+			return;
+		}
+
+		c->Message(Chat::Yellow, fmt::format("Found {} parcels for {}.", parcels.size(), player_name).c_str());
+		for (auto const &p: parcels) {
+			c->Message(
+				Chat::Yellow,
+				fmt::format(
+					"Slot [{:02}] has item id [{:10}] with quantity [{}].",
+					p.second.slot_id,
+					p.second.item_id,
+					p.second.quantity
+				).c_str()
+			);
+		}
+	}
+	if (is_add) {
+		auto to_name  = std::string(sep->arg[2]);
+		auto item_id  = Strings::ToUnsignedInt(sep->arg[3]);
+		auto quantity = Strings::ToUnsignedInt(sep->arg[4]);
+		auto note     = std::string(sep->argplus[5]);
+
+		if (arguments < 2) {
+			SendParcelsSubCommands(c);
+			return;
+		}
+
+		auto send_to_client = ParcelsRepository::GetParcelCountAndCharacterName(
+				database,
+				to_name
+		);
+		if (send_to_client.at(0).character_name.empty()) {
+			c->MessageString(Chat::Yellow, CANT_FIND_PLAYER, to_name.c_str());
+			return;
+		}
+
+		auto next_slot = c->FindNextFreeParcelSlot(send_to_client.at(0).character_name);
+		if (next_slot == INVALID_INDEX) {
+			c->Message(
+				Chat::Yellow,
+				fmt::format(
+					"Unfortunately, {} cannot accept any more parcels at this time. Please try again later.",
+					send_to_client.at(0).character_name
+				).c_str()
+			);
+			return;
+		}
+
+		if (item_id == PARCEL_MONEY_ITEM_ID) {
+			if (quantity > INT32_MAX) {
+				c->Message(
+					Chat::Yellow,
+					"Your quantity of {} copper pieces was too large.  Set to max quantity of {}.",
+					quantity,
+					INT32_MAX
+				);
+				quantity = INT32_MAX;
+			}
+
+			auto item = database.GetItem(PARCEL_MONEY_ITEM_ID);
+			if (!item) {
+				c->Message(Chat::Yellow, "Could not find item with id {}", item_id);
+				return;
+			}
+
+			auto inst = database.CreateItem(item, 1);
```

Comment:
`database.CreateItem` creates a new item instance on the heap and gives the caller ownership so this needs to be deleted to avoid a memory leak when done with it. Suggest just wrapping it in a unique_ptr

Reply:
As noted above, converted to unique_ptr.